source/particle/property/elastic_tensor_decomposition.cc

```diff
@@ -0,0 +1,604 @@
+        projection_matrix_tetra_to_hexa[5][1] = 1./(4.*std::sqrt(2.0));
+        projection_matrix_tetra_to_hexa[8][0] = 0.25;
+        projection_matrix_tetra_to_hexa[8][1] = 0.25;
+        projection_matrix_tetra_to_hexa[8][5] = -1/(2*std::sqrt(2.0));
+
+
+        // projection_matrix_hexa_to_iso;
+        projection_matrix_hexa_to_iso[0][0] = 3./15.;
+        projection_matrix_hexa_to_iso[0][1] = 3./15.;
+        projection_matrix_hexa_to_iso[0][2] = 3./15.;
+        projection_matrix_hexa_to_iso[1][1] = 3./15.;
+        projection_matrix_hexa_to_iso[1][2] = 3./15.;
+        projection_matrix_hexa_to_iso[2][2] = 3./15.;
+        projection_matrix_hexa_to_iso[3][3] = 4./15.;
+        projection_matrix_hexa_to_iso[3][4] = 4./15.;
+        projection_matrix_hexa_to_iso[3][5] = 4./15.;
+        projection_matrix_hexa_to_iso[4][4] = 4./15.;
+        projection_matrix_hexa_to_iso[4][5] = 4./15.;
+        projection_matrix_hexa_to_iso[5][5] = 4./15.;
+        projection_matrix_hexa_to_iso[6][6] = 1./5.;
+        projection_matrix_hexa_to_iso[6][7] = 1./5.;
+        projection_matrix_hexa_to_iso[6][8] = 1./5.;
+        projection_matrix_hexa_to_iso[7][7] = 1./5.;
+        projection_matrix_hexa_to_iso[7][8] = 1./5.;
+        projection_matrix_hexa_to_iso[8][8] = 1./5.;
+
+        projection_matrix_hexa_to_iso[0][3] = std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[0][4] = std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[0][5] = std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[1][3] = std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[1][4] = std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[1][5] = std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[2][3] = std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[2][4] = std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[2][5] = std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[0][6] = 2./15.;
+        projection_matrix_hexa_to_iso[0][7] = 2./15.;
+        projection_matrix_hexa_to_iso[0][8] = 2./15.;
+        projection_matrix_hexa_to_iso[1][6] = 2./15.;
+        projection_matrix_hexa_to_iso[1][7] = 2./15.;
+        projection_matrix_hexa_to_iso[1][8] = 2./15.;
+        projection_matrix_hexa_to_iso[2][6] = 2./15.;
+        projection_matrix_hexa_to_iso[2][7] = 2./15.;
+        projection_matrix_hexa_to_iso[2][8] = 2./15.;
+        projection_matrix_hexa_to_iso[3][6] = -std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[3][7] = -std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[3][8] = -std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[4][6] = -std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[4][7] = -std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[4][8] = -std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[5][6] = -std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[5][7] = -std::sqrt(2.0)/15.;
+        projection_matrix_hexa_to_iso[5][8] = -std::sqrt(2.0)/15.;
```

Comment:
If all of these are hardcoded numbers, make them `static constexpr Tensor<...>` outside of this class. That way the compiler can already optimize their use.